kolibri/core/assets/src/views/k-checkbox/index.vue

```diff
@@ -0,0 +1,191 @@
+<template>
+
+  <div
+    class="k-checkbox-container"
+    :class="{ 'k-checkbox-disabled': disabled }"
+    @click.prevent="toggleCheck"
+  >
+    <div class="tr">
+
+      <div class="k-checkbox" :class="{ 'k-checkbox-active': isActive }">
+        <input
+          type="checkbox"
+          class="k-checkbox-input"
+          :name="name"
+          :checked="isCurrentlyChecked"
+          :indeterminate.prop="isCurrentlyIndeterminate"
+          :disabled="disabled"
+          @click.stop="toggleCheck"
+          @focus="isActive = true"
+          @blur="isActive = false"
+        >
+
+        <mat-svg
+          v-if="isCurrentlyIndeterminate"
+          category="toggle"
+          name="indeterminate_check_box"
+          class="k-checkbox-indeterminate"
+        />
+        <mat-svg
+          v-else-if="!isCurrentlyIndeterminate && isCurrentlyChecked"
+          category="toggle"
+          name="check_box"
+          class="k-checkbox-checked"
+        />
+        <mat-svg
+          v-else
+          category="toggle"
+          name="check_box_outline_blank"
+          class="k-checkbox-unchecked"
+        />
+
+      </div>
+
+      <label v-if="label" :for="name" class="k-checkbox-label">{{ label }}</label>
+
+    </div>
+  </div>
+
+</template>
+
+
+<script>
+
+  /**
+    * A checkbox
+    */
+  export default {
+    name: 'k-checkbox',
+    model: {
+      prop: 'checked',
+      event: 'change',
+    },
+    props: {
+      /**
+       * Name attribute
+       */
+      name: {
+        type: String,
+        required: false,
+      },
```

Comment:
I don't think that `name` is being used correctly. We seem to often use it for labeling/a11y purposes, but it's actually only meant for programmatically referencing the input box and creating groups (e.g. of radio buttons).

It's actually not clear to me that we need to expose `name` as a prop at all. On the other hand, we're leaving ourselves open to 'unlabeled' checkboxes.

----

Proposed refactor:

* remove `name` prop
* make `label` a required prop
* add a boolean `showLabel` prop
* when `showLabel === false`, add `:title="label"` to the checkbox to provide a tooltip, and set `:class="'visuallyhidden'"` on the `<label>` element.